### PR TITLE
Enforce additional error-prone checks

### PR DESCRIPTION
**Pull request by schlosna**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -70,24 +70,98 @@ allprojects {
...
-                '-Xep:PreferSafeLoggableExceptions:ERROR',
-                '-Xep:PreferSafeLoggingPreconditions:ERROR',
-                '-Xep:ReturnMissingNullable:ERROR',
-                '-Xep:Slf4jLogsafeArgs:ERROR',
-                '-Xep:SwitchStatementDefaultCase:ERROR',
-                '-Xep:UnnecessaryParentheses:ERROR',
-                '-Xep:UnusedVariable:ERROR',
             ]
+
+            def warningsToPromote = [
```
Comment (carterkozak) [inline build.gradle]: Should we use the list from errorprone?
https://github.com/google/error-prone/blob/3d225c1eec6d8dada11730c9e9199b710ea26495/core/src/main/java/com/google/errorprone/scanner/BuiltInCheckerSuppliers.java#L593

File: build.gradle [hunk 2]
```diff
@@ -70,24 +70,98 @@ allprojects {
...
+
+            if (generateErrorPronePatch) {
+                def patchChecks = [
+                    "PreferSafeLoggableExceptions",
+                    "PreferCollectionTransform",
+                    "PreferListsPartition",
+                ]
+
+                logger.info("Generating error-prone.patch for checks {} apply with:\n"
+                        + "    find . -maxdepth 3 -type f -name 'error-prone.patch' \\ \n"
```
Comment (carterkozak) [inline build.gradle:109]: I think `-maxdepth 2` is sufficient here, though it doesn't make much difference. Some day I'll get around to writing a proper gradle task for these.

**Review by carterkozak — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (schlosna) [reply build.gradle]: Possibly, will give it a shot adding all non-default enabled. Last time I looked at the non-default ones there were some pathologically slow ones.

File: build.gradle (hunk 2 above)
Comment (schlosna) [reply build.gradle:109]: Yeah, maxdepth 2 works for tritium.

Adding a configurable set of checks and option to generate patch to gradle-Baseline would be useful. Might give that a shot if I can find some time.

**Comment by uschi2000** [comment]
Why not discuss this kind of stuff in the context of Baseline? Then everybody would benefit.

**Comment by schlosna** [comment]
Hey @uschi2000 , I will bring this over to baseline. This PR was just some local changes I had from testing out and applying the checks added in https://github.com/palantir/gradle-baseline/pull/622 and https://github.com/palantir/gradle-baseline/pull/623 against Tritium.

**Comment by uschi2000** [comment]
Cheers!

File: build.gradle (hunk 1 above)
Comment (schlosna) [reply build.gradle]: done, promoted all enabled warnings & disabled checks to treat as errors

**Pull request merged**